test/ceddl.spec.ts

```diff
@@ -24,4 +50,21 @@ describe('mock CEDDL unit tests', () => {
     expect(version).not.be.empty;
   });
 
+
+  it('it should pick data from CEDDL', () => {
+    const { basicDigitalData, ceddlVersion } = CEDDL;
+    expect(basicDigitalData.version).to.not.be.undefined;
+    expect(basicDigitalData.page.pageInfo.pageID).to.not.be.undefined;
+
+    const pick = new PickOperator({
+      name: 'pick',
+      properties: 'version,pageID'
+    })
+    const output = pick.handleData([basicDigitalData]);
+    console.log('output', output);
+    expect(output).to.not.be.null;
+    expect(output![0].version).to.eq(ceddlVersion);
+    expect(output![0].pageID).to.eq(basicDigitalData.page.pageInfo.pageID);
```

Comment:
I may need to clarify/revisit the design of `pick` based on this test.  This test fails because `pageID` won't be found. The reason is because neither `page` or `pageInfo` is provided as a picked property.  So as the picker traverses into child properties, it will encounter `page` and stop traversal because `page` is not listed as a picked property.  This design was intended to guard against picking a property with the same name but in different children.  By providing the parent properties you can better manage the traversal and picked properties.

Reply:
My initial thought was that that pick operator would take dot-path notation like selectors, so the `properties` value would be `version,page.pageInfo.pageID`. The tests made it clear that wasn't the case but perhaps it's something to consider?

Reply:
Yeah, I think we need a more expressive but lightweight syntax: needs to support dot notation, an array index, and maybe a * wildcard - on properties names and `parent.*`.  Defining the grammar and parser outside would get away from these idiosyncracies.

Reply:
Agreed. Since we need the same syntax for rule source selection it seems like building a separate parser/data-selector method makes sense.